legal-api/src/legal_api/services/filings/validations/common_validations.py

```diff
@@ -331,3 +341,58 @@ def validate_foreign_jurisdiction(foreign_jurisdiction: dict,
         msg.append({'error': 'Invalid region.', 'path': f'{foreign_jurisdiction_path}/region'})
 
     return msg
+
+
+def validate_offices_addresses(filing_json: dict, filing_type: str) -> list:
+    """Validate optional fields in office addresses."""
+    msg = []
+    offices_dict = filing_json['filing'][filing_type]['offices']
+    offices_path = f'/filing/{filing_type}/offices'
+    for key, value in offices_dict.items():
+        msg.extend(validate_addresses(value, f'{offices_path}/{key}'))
+    return msg
+
+
+def validate_parties_addresses(filing_json: dict, filing_type: str, key: str = 'parties') -> list:
+    """Validate optional fields in party addresses."""
+    msg = []
+    parties_array = filing_json['filing'][filing_type][key]
+    parties_path = f'/filing/{filing_type}/{key}'
+    for idx, party in enumerate(parties_array):
+        msg.extend(validate_addresses(party, f'{parties_path}/{idx}'))
+    return msg
+
+
+def validate_addresses(
+    addresses: dict,
+    addresses_path: str
+) -> list:
+    """Validate optional fields in addresses."""
+    msg = []
+    for address_type in Address.JSON_ADDRESS_TYPES:
+        if address := addresses.get(address_type):
+            err = _validate_postal_code(address, f'{addresses_path}/{address_type}')
+            if err:
+                msg.append(err)
+    return msg
+
+
+def _validate_postal_code(
+    address: dict,
+    address_path: str
+) -> dict:
+    """Validate that postal code is optional for specified country."""
+    country = address['addressCountry']
+    postal_code = address['postalCode']
```

Comment:
`postalCode` may not be in json
```suggestion
    postal_code = address.get('postalCode')
```